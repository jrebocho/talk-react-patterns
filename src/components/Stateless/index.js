import React from 'react'
import PropTypes from 'prop-types'
import './Stateless.css'

const Stateless = ({ on }) => {
  const boxColor = on ? 'green' : 'red'

  return <div className={`box ${boxColor}`}>{on ? 'ON' : 'OFF'}</div>
}

Stateless.propTypes = {
  on: PropTypes.bool
}

export default Stateless
