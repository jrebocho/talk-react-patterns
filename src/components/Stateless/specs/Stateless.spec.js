import React from 'react'
import { shallow } from 'enzyme'
import Stateless from '../../Stateless'

describe('<Stateless />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Stateless />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is on', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Stateless on={true} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
