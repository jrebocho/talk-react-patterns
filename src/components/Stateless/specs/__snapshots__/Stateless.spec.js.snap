// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stateless /> renders correctly 1`] = `
<div
  className="box red"
>
  OFF
</div>
`;

exports[`<Stateless /> when the component is on renders correctly 1`] = `
<div
  className="box green"
>
  ON
</div>
`;
